src/server/generic_family.cc

```diff
@@ -406,6 +409,21 @@ OpStatus Renamer::UpdateDest(Transaction* t, EngineShard* es) {
     if (!is_prior_list && dest_it->second.ObjType() == OBJ_LIST && es->blocking_controller()) {
       es->blocking_controller()->AwakeWatched(t->db_index(), dest_key);
     }
+    if (es->journal()) {
+      OpArgs op_args = t->GetOpArgs(es);
+      string scratch;
+      // todo insert under multi exec
+      RecordJournal(op_args, "SET"sv, ArgSlice{dest_key, dest_it->second.GetSlice(&scratch)}, 2,
+                    true);
+      if (dest_it->first.IsSticky()) {
+        RecordJournal(op_args, "STICK"sv, ArgSlice{dest_key}, 2, true);
+      }
+      if (dest_it->second.HasExpire()) {
+        auto time = absl::StrCat(src_res_.expire_ts);
+        RecordJournal(op_args, "PEXPIREAT"sv, ArgSlice{time}, 2, true);
+      }
+      RecordJournalFinish(op_args, 2);
+    }
```

Comment:
So you send
1. DEL with txid=T
2. MUTLI(implied by opcode) SET EXPIREAT STICK EXEC with txid=T

In theory, do not DEL and SET have to be in one transaction (i.e. inside one multi)? But we cannot do such atomic operations on replica currently, right?

I'd also suggest assembling the latter command as one, because each RecordJournal puts pressure on the writing and receving side... But there is currently no option, because there is no stick parameter for set, right? Maybe we should add one in the future?

Reply:
The DEL and SET are in one transaction because we shard_count=2 and the same txid. This is the same as multi shard commands which are not inside multi command transaction. Currently the replica just does not executes the transaction if data from all shard is not received i.e it will not execute the SET if the DEL was not received. 
I use the multi just in the SET flow because we want to execute EXPIREAT and STICK in the same transaction, this also does not guarantees atomicity because currently we do the execution of the commands one by one.

We could do optimization for the SET flow for one command which will include the expire info and the stick info, but I am not sure its needed. Rename is probably not common command, and if this is the only flow we have this multi commands this is probably not worth it. Lets implement all the rename flows and see if this is needed